lib/internal/assert/calltracker.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const {
+  Error,
+  SafeSet,
+} = primordials;
+
+const {
+  codes: {
+    ERR_UNAVAILABLE_DURING_EXIT,
+  },
+} = require('internal/errors');
+const AssertionError = require('internal/assert/assertion_error');
+const {
+  validateBoolean,
+  validateObject,
+  validateUint32,
+} = require('internal/validators');
+
+const noop = () => {};
+
+class CallTracker {
+
+  #callChecks = new SafeSet()
+
+  constructor(options = { onExit: false }) {
+    validateObject(options, 'options');
+    const { onExit } = options;
+    validateBoolean(onExit, 'options.onExit');
+    if (onExit) {
+      process.on('exit', () => this.verify());
```

Comment:
By attaching this in the constructor, there isn't a really straightforward way to remove the listener if you change your mind.

Reply:
I'm not sure what you mean by "change your mind"

Reply:
Call `removeListener()`.

Reply:
The way to handle that easily @ConorDavenport would be to include something like...

```js
class CallTracker {

 // ...
 #onExit = function() { this.verify() };

 // ...
 constructor(options = { onExit: false }) {
   // ...
   if (onExit)
     process.on('exit', this.#onExit);
   // ...
  }

  // ...

  removeExitHandler() {
    process.removeListener('exit', this.#onExit)
  }
}
```

Reply:
That kind of goes back to having a one line function for dealing with exit handlers that this constructor option was meant to avoid. It seems like it would be simpler to go back to having the "add exit handler" function which can return the added handler. Then, if someone wants to remove it, they can use the standard APIs for dealing with event emitters.

Reply:
Yeah I think you're right